### ISSUE TITLE
Doesn't survive process deaths

### ISSUE THREAD

**Issue by hannesstruss** [VictorAlbertos/RxActivityResult#9]
Steps to reproduce:

1. Pick "Using activity"
2. Pick "Camera"
3. [Camera started] – Kill app process with ADB
4. Shoot photo

Results in a crash:

```
FATAL EXCEPTION: main
Process: io.victoralbertos.app, PID: 3083
java.lang.RuntimeException: Unable to start activity ComponentInfo{io.victoralbertos.app/rx_activity_result.HolderActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'rx_activity_result.OnResult rx_activity_result.Request.onResult()' on a null object reference
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2625)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2686)
    at android.app.ActivityThread.-wrap12(ActivityThread.java)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1440)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:5969)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:801)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:691)
 Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'rx_activity_result.OnResult rx_activity_result.Request.onResult()' on a null object reference
    at rx_activity_result.HolderActivity.onCreate(HolderActivity.java:33)
    at android.app.Activity.performCreate(Activity.java:6669)
    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2578)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2686) 
    at android.app.ActivityThread.-wrap12(ActivityThread.java) 
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1440) 
    at android.os.Handler.dispatchMessage(Handler.java:102) 
    at android.os.Looper.loop(Looper.java:154) 
    at android.app.ActivityThread.main(ActivityThread.java:5969) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:801) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:691) 
```

**Comment by VictorAlbertos** [comment]
Thanks. I'll take a look. 

But, has this "clinic" test real consequences for a normal use of an Android Application? What  real scenario would be able to mimic this behaviour? The user leaves the app and after some time he comes to the app but -oh my god- android has killed the process?

**Comment by VictorAlbertos** [comment]
And by the way, can post the code related with the subscription event?

**Comment by hannesstruss** [comment]
It's a scenario on low-memory devices, where the system may [kill the process to reclaim memory.](https://developer.android.com/guide/topics/processes/process-lifecycle.html)

No own code, I was using the sample app from this repository. You can also reproduce this by setting the maximum number of background processes to zero in the dev settings.

**Comment by VictorAlbertos** [comment]
Thanks @hannesstruss. I'll take a look.

**Comment by VictorAlbertos** [comment]
I've been trying to recreate the issue but I've not been able to achieve it.

When I try to kill the app process using adb the application just closes, so I'm not able to reproduce the trace that you mentioned previously. 

The command which I execute is:
`adb shell am force-stop io.victoralbertos.app`

**Comment by VictorAlbertos** [comment]
Closing with issue due to lack of feedback.

**Comment by iveyalkin** [comment]
We've got quite a few simillar crashes on production on various devices with 4.x-6.x Android OS, so that case is pretty plausible. I can't reproduce it with neither `limit background processes` not `adb shell am force-stop` as well. However, I've managed to reproduce the crash in a slightly extended way:
1. Limit background processes to 0 via dev settings as suggested previously. 
2. Start your normal sending intent flow with `RxActivityResult`, perhaps, you'd like to pick a new profile picture:
```
// ... prepare 'openGalleryIntent'
RxActivityResult.on(currentActivity)
                .startIntent(openGalleryIntent)
                .subscribe(...)
```
3. Switch to a different process (i.e. check out some new Tweets)
4. Switch back to you app via Overview Screen (this little square button that shows a list of recent tasks)
5. Catch the NPE, as static variable `private static Request request` in `HolderActivity` won't survive process re-creation.

**Comment by VictorAlbertos** [comment]
Thanks @iveyalkin for the detailed steps to reproduce the bug.

I'll check it out as soon as possible.

**Comment by VictorAlbertos** [comment]
I was able to reproduce the bug. Now ActivityHolder performs a defensive null check in order to avoid a null pointer exception. But the fix isn't completely optimal because once the activity has been destroyed the user is going to need to call again the intent. But, I can't think in a way to preserve the state in such hostile scenario. 

Available from version 0.3.5
